airgun/views/module_streams.py

```diff
@@ -0,0 +1,85 @@
+
+from widgetastic.widget import (
+    Table,
+    Text,
+    TextInput,
+    View,
+)
+from widgetastic_patternfly import BreadCrumb
+
+from airgun.views.common import (
+    BaseLoggedInView,
+    SatTab,
+    SatTable,
+)
+from airgun.widgets import Search
+
+
+class CustomSearch(Search):
+    search_field = TextInput(locator=".//input[@role='combobox']")
+    search_button = Text(".//button[text()='Search']")
+
+
+class ModuleStreamView(BaseLoggedInView):
+    """Main Module_Streams view"""
+    title = Text("//h2[contains(., 'Module Streams')]")
+    table = SatTable('.//table', column_widgets={'Name': Text("./a")})
+
+    search_box = CustomSearch()
+
+    def search(self, query):
+        """Perform search using search box on the page and return table
+        contents.
+
+        :param str query: search query to type into search field. E.g.
+            ``name = "bar"``.
+        :return: list of dicts representing table rows
+        :rtype: list
+        """
+        self.search_box.search(query)
+        return self.table.read()
+
+    @property
+    def is_displayed(self):
+        """The view is displayed when it's title exists"""
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+
+class ModuleStreamDetailsTable(Table):
+
+    @property
+    def headers(self):
+        """Explicitly returns the headers name and value"""
+        return 'name', 'value'
+
+    def read(self):
+        """Transform rows to a dict {property_name: property_value ...}."""
+        properties = super().read()
+        return {prop['name']: prop['value']
+                for prop in properties if prop['value']}
```

Comment:
Whole `ModuleStreamDetailsTable` seems to be modelled after `DiscoveredHostDetailsTable`, but you only took two methods, while the other one had more fields and also their own Row and Column classes. Could you explain your reasoning here?

On related note, if ModuleStreamDetailsTable is pretty much the same as DiscoveredHostDetailsTable, I think we should rename it to "KeyValueTable" or something and move it out to `widgets`. Possibly not as part of this PR - feel free to open issue to discuss that.

Reply:
ModuleStreamDetailsTable is almost the same as DiscoveredHostDetailsTable without having headers, that was the I don't require to add other methods.

About making as "KeyValueTable" into widgets is a better idea as well 

@ldjebran, @oshtaier @al what you guys think?

Reply:
Think, We may add it to widgets module, with a proper naming , but do not think it's a requirement nor a blocker.